### DISCUSSION TITLE
Looping wagmi/cli generated hooks

### DISCUSSION THREAD

**Discussion by endresol** [wevm/wagmi#3430] [Q&A]
Hello all.

I have used wagmi/cli to generate hooks for an inherited ERC-721 contract that is already deployed and minted out.

From the generated code I have one hook useContractGetBalance() that returns the number of NFT a user holds
Then I have a useContractTokenOfOwnerByIndex(int) that returns the tokenID of one of the NFTs the user holds. 
And then lastly, I have useContractGetTokenValue(bitint) that give me one of three values of the token.

I want to find the token with the highest value that the user owns.

Normally I would get the balance, then loop through and get the TokenOfOwnerByIndex to get the tokenId and then use the GetTokenValue to get the value and validate if this is the higher value. 
If it is the max value I don't need to do anything more, if no I will check the next value...
```
// pseudo code
let maxValue = 0;
const balance = await contract.getBalance(address);
 for (let index = 0; index < balance; index++) {
      const token = await contract.TokenOfOwnerByIndex(index);
      const value = await contract.getTokenValue(token);
      if (value > maxValue) { maxValue = value};
    }
return maxValue;
```

OK, hope somebody is still with me.

Using hooks in loop is not a good idea, so how should I solve this?

**Comment by art-eth** [comment]
Use `getContract` hook and call functions using `read` property.

Example:
```
const publicClient = usePublicClient()
const erc721 = getContract({
    address: '0x....',
    abi: erc721ABI,
    publicClient,
  })
const jsonUri = await erc721.read.tokenURI([token])
```
